src/renderer/components/dock/upgrade-chart.tsx

```diff
@@ -126,6 +121,11 @@ export class UpgradeChart extends React.Component<Props> {
           submittingMessage={_i18n._(t`Updating..`)}
           controls={controlsAndInfo}
         />
+        <EditorPanel
```

Comment:
Not saying that this should be done in this PR, but it does seem like the dock code repeats itself quite often. Do you think it would be worth making this more DRY?

Reply:
Definitely, it's always a room for improvements.